### ISSUE TITLE
destroy! on an ActiveRecord::Collection 

### ISSUE THREAD

**Issue by matt297** [ActsAsParanoid/acts_as_paranoid#82]
**Summary:** Calling `.destroy!` on an `ActiveRecord::Collection` results in unexpected behaviour. According to the AR documentation, `.destroy!` is not implemented for collections and should result in an undefined method exception.

**Use case:** A has_many through relationship, trying to delete a join row. Using `gem 'rails', '~> 5.0.0'` and `gem 'acts_as_paranoid', '~> 0.5.0'`.

**Details:**
```ruby
class Subscription < ApplicationRecord
  acts_as_paranoid
  has_many :subscription_features, dependent: :destroy
  has_many :features, through: :subscription_features
end

class Feature < ApplicationRecord
  acts_as_paranoid
  has_many :subscription_features, dependent: :destroy
  has_many :subscriptions, through: :subscription_features
end

class SubscriptionFeature < ApplicationRecord
  acts_as_paranoid
  belongs_to :subscription
  belongs_to :feature
end
```

If you want to remove a feature from a given subscription, using `.destroy` behaves as expected:
```ruby
sub = Subscription.create
feat = Feature.create
sub.features << feat

# This works! Feature is left intact, only the join row is soft-deleted
sub.features.destroy(feat)
=> (0.1ms)  BEGIN
      SubscriptionFeature Load (0.6ms)  SELECT `subscription_features`.* FROM `subscription_features` WHERE `subscription_features`.`subscription_id` = 1 AND `subscription_features`.`feature_id` = 1
      SQL (2.0ms)  DELETE FROM `subscription_features` WHERE `subscription_features`.`id` = 1
   (0.4ms)  COMMIT
```

But when you try doing the same thing using `.destroy!`, it soft-deletes the feature itself, and doesn't do anything at all to the join row:
```ruby
sub = Subscription.create
feat = Feature.create
sub.features << feat

sub.features.destroy!(feat)
=> SQL (6.1ms)  DELETE FROM `features` WHERE `features`.`id` IN (SELECT id FROM (SELECT DISTINCT `features`.`id` FROM `features` INNER JOIN `subscription_features` ON `features`.`id` = `subscription_features`.`feature_id` WHERE `features`.`deleted_at` IS NULL AND `subscription_features`.`deleted_at` IS NULL AND `subscription_features`.`subscription_id` = 2 AND `features`.`id` = 2) __active_record_temp)
```

Without `acts_as_paranoid` installed, doing `.destroy!` as shown above results in:
```
undefined method `destroy!' for #<Feature::ActiveRecord_Associations_CollectionProxy:0x007fe9e8e836e8>
```

Not sure if this difference in behaviour is intended, or is perhaps a bug. Either way it's a bit confusing to run into and could lead to some nasty headaches, so probably worth addressing 😄

**Comment by mvz** [comment]
The `destroy!` method on collections has been deprecated and will be removed in version 0.8.0.

**Comment by mvz** [comment]
The deprecated `#destroy` method has now been removed. This change will be released in version 0.8.0.
